examples/plot_metrics_with_error.py

```diff
@@ -0,0 +1,163 @@
+# %%
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+================================
+Plotting Metrics with Error Bars
+================================
+"""
+# %%
+# Load and preprocess the data set
+# --------------------------------
+# We download the data set using `fetch_adult` function in `fairlearn.datasets`.
+# We start by importing the various modules we're going to use:
+#
+
+import numpy as np
+import pandas as pd
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.metrics import recall_score
+from sklearn.datasets import fetch_openml
+from sklearn.metrics import confusion_matrix
+from fairlearn.metrics import MetricFrame, ErrorPlotter
+
+
+data = fetch_openml(data_id=1590, as_frame=True)
+X = pd.get_dummies(data.data)
+y_true = (data.target == '>50K') * 1
+sex = data.data['sex']
+
+# %%
+# Error Metrics
+# -------------
+# We have many different choices for error metrics. In this notebook we'll just be using a
+# Normal approximation interval (symmetric) and a Wilson score interval (asymmetric).
+#
+# https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Wilson_score_interval
+# https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Normal_approximation_interval
+#
+#
+
+z_score = 1.959964
+digits = 4
+
+
+def recall_normal_err(y_t, y_p):
+    assert len(y_t) == len(y_p)
+    tn, fp, fn, tp = confusion_matrix(y_t, y_p).ravel()
+    error = compute_error_metric(tp/(tp+fn), tp + fn, z_score=z_score)
+    return error
+
+
+def wilson(p, n, digits=digits, z=z_score):
+    """ Returns lower and upper bound """
+    denominator = 1 + z**2/n
+    centre_adjusted_probability = p + z*z / (2*n)
+    adjusted_standard_deviation = np.sqrt((p*(1 - p) + z*z / (4*n)))/np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z*adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z*adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+def compute_error_metric(metric_value, sample_size, z_score):
+    return z_score*np.sqrt(metric_value*(1.0-metric_value))/np.sqrt(sample_size)
+
+
+def recall_wilson_lower_bound(y_t, y_p):
```

Comment:
Rather than return these separately (which is going to wind up computing the confusion matrix twice), could we have a single `recall_wilson_bound()` and just store the tuple as the metric?